### ISSUE TITLE
InfluxDBResultMapper not thread safe?

### ISSUE THREAD

**Issue by Schntzlbrmpf** [influxdata/influxdb-java#430]
I am refering to this part of the documentation:

InfluxDBResultMapper resultMapper = new InfluxDBResultMapper(); // thread-safe - can be reused
List<Cpu> cpuList = resultMapper.toPOJO(queryResult, Cpu.class);

The problem is when I perform a multi-threaded mapping similar to this part of the code in several threads in parallel:

List<Cpu> cpuList = resultMapper.toPOJO(queryResult, Cpu.class);

I do get the following exception:

org.influxdb.InfluxDBMapperException: java.lang.IllegalAccessException: Class org.influxdb.impl.InfluxDBResultMapper can not access a member of class org.influx.test.Cpu with modifiers "private"
        at org.influxdb.impl.InfluxDBResultMapper.parseSeriesAs(InfluxDBResultMapper.java:184)
        at org.influxdb.impl.InfluxDBResultMapper.lambda$null$2(InfluxDBResultMapper.java:100)

The problem is most likely that the first thread did already set the accessibility of a field back to its original value when the second thread tries to write the field.

**Comment by fmachado** [comment]
:-O

I'll fix the `IllegalAccessException` caused by the concurrent modification on the 'accessible' flag.

I'm sure we may see even performance improvements by checking first if the member is accessible or not, before setting it to true. Currently we are wasting time [here](https://github.com/influxdata/influxdb-java/blob/master/src/main/java/org/influxdb/impl/InfluxDBResultMapper.java#L208) because `setAccessible` is [invoking](http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/lang/reflect/AccessibleObject.java#l126) `System.getSecurityManager()` every time.

Currently I'm returning the 'accessible' flag to the previous state but I don't think there is any benefit from doing it so I'll remove this approach.

Thanks @Schntzlbrmpf !

**Comment by Schntzlbrmpf** [comment]
Hi Fernando,

The way I see it, it might be the best solution to: 

1. either allow the @Column annotation on getters and setters as well which by definition must be public
2. or have the annotation processor check if there are getters and setters on the POJO and use them instead of the Field - here you could fail if there are no such methods

The main problem with using the field directly is that it must be public if we want a fast mapping. If we could make the mapper call the (public) accessor methods instead, we might get a considerable performance increase and at the same time keep a clean interface even to our POJOs.

Best regards and thank you,
Andreas.

**Comment by fmachado** [comment]
@Schntzlbrmpf The current implementation is making changes (by setting the *accessible* flag to true) that were supposed to be permanent so, there is no reason to revert it. I did this initially because (1) I could not identify any collateral affect and (2) it's more simple and performant than both solutions that you proposed (remember: I'm doing not per instance but at class level).

Take a look at the PR that I've just added here. Am I missing something?
